src/api/java/baritone/api/utils/command/BaritoneChatControl.java

```diff
@@ -0,0 +1,225 @@
+            event.cancel();
+        }
+    }
+
+    private void logRanCommand(String command, String rest) {
+        if (settings.echoCommands.value) {
+            String msg = command + rest;
+            String toDisplay = settings.censorRanCommands.value ? command + " ..." : msg;
+
+            ITextComponent component = new TextComponentString(String.format("> %s", toDisplay));
+            component.getStyle()
+                    .setColor(TextFormatting.WHITE)
+                    .setHoverEvent(new HoverEvent(
+                            HoverEvent.Action.SHOW_TEXT,
+                            new TextComponentString("Click to rerun command")
+                    ))
+                    .setClickEvent(new ClickEvent(
+                            ClickEvent.Action.RUN_COMMAND,
+                            FORCE_COMMAND_PREFIX + msg
+                    ));
+
+            logDirect(component);
+        }
+    }
+
+    public boolean runCommand(String msg) {
+        if (msg.trim().equalsIgnoreCase("damn")) {
+            logDirect("daniel");
+            return false;
+        } else if (msg.trim().equalsIgnoreCase("orderpizza")) {
+            try {
+                ((IGuiScreen) mc.currentScreen).openLink(new URI("https://www.dominos.com/en/pages/order/"));
+            } catch (NullPointerException | URISyntaxException ignored) {}
+
+            return false;
+        }
+
+        if (msg.isEmpty()) {
+            msg = "help";
+        }
+
+        Pair<String, List<CommandArgument>> pair = CommandExecution.expand(msg);
+        String command = pair.first();
+        String rest = msg.substring(pair.first().length());
+        ArgConsumer argc = new ArgConsumer(pair.second());
+
+        if (!argc.has()) {
+            Settings.Setting setting = settings.byLowerName.get(command.toLowerCase(Locale.US));
+
+            if (setting != null) {
+                logRanCommand(command, rest);
+
+                if (setting.getValueClass() == Boolean.class) {
+                    CommandManager.execute(String.format("set toggle %s", setting.getName()));
+                } else {
+                    CommandManager.execute(String.format("set %s", setting.getName()));
+                }
+
+                return true;
+            }
+        } else if (argc.hasExactlyOne()) {
+            for (Settings.Setting setting : settings.allSettings) {
+                if (setting.getName().equals("logger")) {
+                    continue;
+                }
+
+                if (setting.getName().equalsIgnoreCase(pair.first())) {
+                    logRanCommand(command, rest);
+                    CommandManager.execute(String.format("set %s %s", setting.getName(), argc.getString()));
+                    return true;
+                }
+            }
+        }
+
+        CommandExecution execution = CommandExecution.from(pair);
+
+        if (isNull(execution)) {
```

Comment:
why is this a thing. can you please just use `== null` like a normal person this is sickening.